### PR TITLE
include BAP regression test

### PR DESCRIPTION
**Pull request by HansVRP**
Included a BAP regression test. 

Needs to be properly scaled credit wise once cdse is stable again

### CODE REVIEW

**Comment by HansVRP** [comment]
@jdries can we add Vincent for the repo for a first review. I compartmentalized the BAP workflow and would like him to review if it was done correctly.

File: test_regression.py [hunk 1]
```diff
@@ -142,3 +152,69 @@ def test_mask_scl(auth_connection, tmp_path):
...
+    # Fetch 100km x 100 km area from geojson
+    spatial_geometries = spatial_geometries.to_crs(epsg=4326)
+    area = eval(spatial_geometries.to_json())
+
+    # Get the spectral bands of interest
+    cube = auth_connection.load_collection(
+        collection_id,
+        temporal_extent = temporal_extent,
+        bands = ["B02", "B03","B04"],
+        max_cloud_cover=70
```
Comment (VincentVerelst) [inline test_regression.py]: Is there a reason you apply a different max_cloud_cover for the RGB bands than for the SCL band?

File: utils_BAP.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    _log.info(f'aggregating rank score')
+    # Aggregate scores
+    weights = [1, 0.8, 0.5]
+    score = (weights[0] * dtc_score +
+             weights[1] * date_score +
+             weights[2] * coverage_score) / sum(weights)
+    return score
+
+
+def create_rank_mask(score: openeo.DataCube) -> openeo.DataCube:
```
Comment (VincentVerelst) [inline utils_BAP.py]: You have create_rank_mask defined twice.

File: test_regression.py (hunk 1 above)
Comment (HansVRP) [reply test_regression.py]: I believe it is done so in the community example, will standardize

File: utils_BAP.py (hunk 2 above)
Comment (HansVRP) [reply utils_BAP.py]: will remove

**Pull request merged**